### ISSUE TITLE
problem with future 

### ISSUE THREAD

**Issue by saikatghosh90** [ourownstory/neural_prophet#1326]
cant create future for 2 yrs, for solar radiation forecasting. 

import pandas as pd
import numpy as np
from neuralprophet import NeuralProphet, set_log_level

data = pd.read_csv("usa.csv")
data["datetime"] = data["Date"] + " " + data["Time"]
data["datetime"] = pd.to_datetime(data["datetime"], format="%d.%m.%Y %H:%M")

sf_pv_df = pd.DataFrame(columns=['ds', 'y'])

sf_pv_df['ds']= data["datetime"]
sf_pv_df['y'] = data["GHI"]

m = NeuralProphet(
    growth="off",
    yearly_seasonality=False,
    weekly_seasonality=False,
    daily_seasonality=False,
    n_lags=3 * 24,
    num_hidden_layers=6,
    d_hidden=8,
    learning_rate=0.0001)
df = sf_pv_df.copy(deep=True)

#%%

a = sf_pv_df["y"].values[1:] - sf_pv_df["y"].values[:-1]
df["I"] = np.append(0, a)

#%%
df_train, df_test = m.split_df(df, freq="H", valid_p=0.10)
m = m.add_lagged_regressor("I", normalize="standardize")
metrics = m.fit(df_train, freq="H", validation_df=df_test, progress="plot")
metrics.tail(1)
m = m.highlight_nth_step_ahead_of_each_forecast(1)

#%%
future = m.make_future_dataframe(df, periods=8760, n_historic_predictions=len(df))

forecast = m.predict(future)

#forecast.to_csv("predicted_values.csv", index=False)
fig = m.plot(forecast)
fig_comp = m.plot_components(forecast)

fig_prediction = m.plot(forecast[:])

# Export predicted values to CSV
forecast.to_csv("predicted_values.csv", index=False)

**Comment by leoniewgnr** [comment]
@saikatghosh90 thanks for raising this! what is the error message you get?

**Comment by saikatghosh90** [comment]
didnt got any error message, the code is not creating a df for future 2 yrs, its just creating a df for future 2 -10 values (hourly). But as far as i remember, prior to the update, code was creating a df for the desired period , 1 or 2 yr. 
Also you can check at your end that this example page is not even properly now - https://neuralprophet.com/how-to-guides/application-examples/energy_solar_pv.html

**Comment by ourownstory** [comment]
Hi @saikatghosh90 
The model parameters that you show in your example code include `n_lags=3 * 24`.
When you use auto-regression (AR), only the specified `n_forecast` next steps can be predicted at once. 
In this case, as  `n_lags` specified, but `n_forecast` is unspecified, it is set to `n_forecast=1`.
When using AR on hourly data, a forecast horizon of two years seems too large. 
You may want to reduce your horizon (by setting `n_forecast`) to a few hours or days. 
Alternatively, you can not use AR (setting `n_lags=0`) and then you can extrapolate the model to larger horizons, potentially at the cost of accuracy.

**Comment by leoniewgnr** [comment]
Hi @saikatghosh90, if you were referring to the missing plots in the docs. I fixed them in #1328, thanks for raising this. Let me know if you find something else!

**Comment by leoniewgnr** [comment]
@saikatghosh90 does this answer your question?
If not, please feel free to raise a question in our slack channel.
